src/v/utils/tests/object_pool_test.cc

```diff
@@ -0,0 +1,122 @@
+
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "seastarx.h"
+#include "utils/object_pool.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <boost/test/tools/old/interface.hpp>
+
```

Comment:
Might be worthwhile doing test that you can construct & use the pool with a move-only object.

Reply:
Added a unit test that uses a move only object